src/main/kotlin/me/lensvol/blackconnect/config/sections/LocalDaemonSection.kt

```diff
@@ -0,0 +1,246 @@
+package me.lensvol.blackconnect.config.sections
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.invokeLater
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.components.service
+import com.intellij.openapi.fileChooser.FileChooserDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.TextBrowseFolderListener
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
+import com.intellij.ui.IdeBorderFactory
+import com.intellij.util.ui.FormBuilder
+import com.intellij.util.ui.JBUI
+import me.lensvol.blackconnect.BlackdExecutor
+import me.lensvol.blackconnect.ExecutionResult
+import me.lensvol.blackconnect.config.DEFAULT_BLACKD_PORT
+import me.lensvol.blackconnect.settings.BlackConnectGlobalSettings
+import me.lensvol.blackconnect.settings.BlackConnectProjectSettings
+import me.lensvol.blackconnect.ui.AdditionalInformationDialog
+import me.lensvol.blackconnect.ui.disableContents
+import me.lensvol.blackconnect.ui.enableContents
+import java.awt.Component
+import java.awt.Dimension
+import java.awt.GridBagConstraints
+import java.awt.GridBagLayout
+import javax.swing.Box
+import javax.swing.BoxLayout
+import javax.swing.JButton
+import javax.swing.JCheckBox
+import javax.swing.JComponent
+import javax.swing.JLabel
+import javax.swing.JPanel
+import javax.swing.JSpinner
+import javax.swing.JTextField
+import javax.swing.SpinnerNumberModel
+import javax.swing.event.DocumentEvent
+import javax.swing.event.DocumentListener
+import kotlin.concurrent.thread
+
+class LocalDaemonSection(val project: Project) : ConfigSection(project) {
+    private val startLocalServerCheckbox = JCheckBox("Start local blackd instance when plugin loads")
+    private val remotePortModel = SpinnerNumberModel(DEFAULT_BLACKD_PORT, 1, 65535, 1)
+    private val bindOnHostnameText = JTextField("127.0.0.1")
+    private val localPortSpinner = JSpinner(remotePortModel)
+    private val blackdExecutableChooser = TextFieldWithBrowseButton().apply {
+        val fileChooserDescriptor = FileChooserDescriptor(
+            true,
+            false,
+            false,
+            false,
+            false,
+            false
+        )
+        addBrowseFolderListener(TextBrowseFolderListener(fileChooserDescriptor))
+    }
+    private val startDaemonButton = JButton("Start", AllIcons.Actions.Execute)
+    private val stopDaemonButton = JButton("Stop", AllIcons.Actions.Suspend)
+
+    private val blackdExecutor = service<BlackdExecutor>()
+
+    private val localServerPanel = createLocalServerPanel()
+
+    init {
+        localPortSpinner.editor = JSpinner.NumberEditor(localPortSpinner, "#")
+
+        localServerPanel.disableContents()
+
+        installUiListeners()
+    }
+
+    override val panel: JPanel = createPanel()
+
+    private fun createPanel(): JPanel {
+        return JPanel().apply {
```

Comment:
The Black settings configurable UI has started to become more complicated indeed, as you've mentioned on Twitter. For future UI changes / UI refactoring we might consider [Kotlin UI DSL](https://plugins.jetbrains.com/docs/intellij/kotlin-ui-dsl.html) I mentioned earlier.